database/helpers/BaseCachedEntityTableMigration.php

```diff
@@ -0,0 +1,14 @@
+<?php
+
+namespace SyncTools\Database\Helpers;
+
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Support\Facades\Config;
+
+class BaseCachedEntityTableMigration extends Migration
+{
+    public function getConnection(): ?string
+    {
+        return Config::get('pgsql-connection.sync.name', $this->connection);
+    }
```

Comment:
@DanBondarenko You are correct, the main idea was to run all migrations using the `admin` connection like this: `php artisan migrate --database=pgsql_main` but during the implementation, I found that when you try to create a table in the schema different from the `database.pgsql.search_path` (default schema name) it will switch the default schema of the connection that was used. That's why I decided to run the cached table migrations using the sync user as it has the privilege to do it inside the corresponding schema.

Reply:
A-ha, I see. Thank you for the explanation. :)